src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java

```diff
@@ -29,26 +29,44 @@
     return Or.of(newChildren);
   }
 
-  public static <K> Expression<K> collapseToPOS(Or<K> or, And<K> internalAnd, Expression<K> omitFromAnd){
+  public static <K> Expression<K> stripNegation(And<K> and, Or<K> internalOr, Expression<K> omitFromOr){
+    Expression<K>[] childrenNew = ExprUtil.allExceptMatch(and.expressions, internalOr);
+    List<Expression<K>> newChildren = Lists.newArrayList(childrenNew);
+
+    List<Expression<K>> orOthers = Lists.newArrayList();
+    for (Expression<K> orChild : internalOr.expressions) {
+
+      if(!orChild.equals(omitFromOr)) {
+        orOthers.add(orChild);
+      }
+
+    }
+
+    newChildren.add(Or.of(orOthers));
+
+    return And.of(newChildren);
+  }
+
+  public static <K> Expression<K> stripNegation(Or<K> or, And<K> internalAnd, Expression<K> omitFromAnd){
     Expression<K>[] childrenNew = ExprUtil.allExceptMatch(or.expressions, internalAnd);
-    List<Expression<K>> newChildren = Lists.newArrayList();
+    List<Expression<K>> newChildren = Lists.newArrayList(childrenNew);
 
+    List<Expression<K>> andOthers = Lists.newArrayList();
     for (Expression<K> andChild : internalAnd.expressions) {
-      if(!andChild.equals(omitFromAnd)) {
-        List<Expression<K>> orOthers = Lists.newArrayList();
-        ExprUtil.addAll(orOthers, childrenNew);
-        orOthers.add(andChild);
 
-        newChildren.add(Or.of(orOthers));
+      if(!andChild.equals(omitFromAnd)) {
+        andOthers.add(andChild);
       }
 
     }
 
-    return And.of(newChildren);
+    newChildren.add(And.of(andOthers));
+
+    return Or.of(newChildren);
   }
 
   public static <K> Expression<K>[] allExceptMatch(Collection<Expression<K>> exprs, Set<? extends Expression<K>> omit){
-    Set<Expression<K>> andTerms = new HashSet<Expression<K>>();
+    Set<Expression<K>> andTerms = new LinkedHashSet<Expression<K>>();
```

Comment:
The linkedhashset will preserve the order, but is there really a point to this, since we are already moving the new term to the end?  Or does this help tests pass?

Reply:
Yeah. We need the tests to be deterministic.